Plugin load error starting with version 2.0.0 (TeamCity 2019.1.5)

Hello.
After updating to the most recent version of the plugin (2.1.0) I'm getting a `Failed to initialize Spring Context` error.  
I've tried installing some previous versions and it seems, that this problem appear in v2.0.0 release.

Unfortunately, I'm unable to upgrade TC due to expired agents licenses.
Is there a minimal required version for TC that the plugin supports?

Here is the full error log from TC plugins page:
```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.fpd.teamcity.slack.controllers.ConfigController#0' defined in Byte array resource [plugin: Slack Integration#slackIntegration-server-1.0.0.jar!/META-INF/build-server-plugin-slackIntegration.xml]: Unsatisfied dependency expressed through constructor parameter 3; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.fpd.teamcity.slack.SlackGateway#0' defined in Byte array resource [plugin: Slack Integration#slackIntegration-server-1.0.0.jar!/META-INF/build-server-plugin-slackIntegration.xml]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.fpd.teamcity.slack.SlackGateway]: Constructor threw exception; nested exception is java.lang.NullPointerException
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:749)
at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:189)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1193)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1095)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761)
at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543)
at jetbrains.buildServer.plugins.spring.SpringPluginLoader.pluginClassesLoaded(SpringPluginLoader.java:114)
at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at jetbrains.buildServer.util.EventDispatcher$3.run(EventDispatcher.java:131)
at jetbrains.buildServer.util.NamedThreadFactory.executeWithNewThreadName(NamedThreadFactory.java:76)
at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:125)
at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:75)
at com.sun.proxy.$Proxy25.pluginClassesLoaded(Unknown Source)
at jetbrains.buildServer.plugins.PluginManagerImpl$5.visitPlugin(PluginManagerImpl.java:417)
at jetbrains.buildServer.plugins.PluginsCollection$1.run(PluginsCollection.java:97)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.fpd.teamcity.slack.SlackGateway#0' defined in Byte array resource [plugin: Slack Integration#slackIntegration-server-1.0.0.jar!/META-INF/build-server-plugin-slackIntegration.xml]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.fpd.teamcity.slack.SlackGateway]: Constructor threw exception; nested exception is java.lang.NullPointerException
at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:279)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1193)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1095)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066)
at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:835)
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:741)
... 28 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.fpd.teamcity.slack.SlackGateway]: Constructor threw exception; nested exception is java.lang.NullPointerException
at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:154)
at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:122)
at jetbrains.buildServer.spring.InstantiationStrategySelector$1.instantiate(InstantiationStrategySelector.java:77)
at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:271)
... 41 more
Caused by: java.lang.NullPointerException
at com.slack.api.util.http.SlackHttpClient.buildOkHttpClient(SlackHttpClient.java:59)
at com.slack.api.util.http.SlackHttpClient.buildSlackHttpClient(SlackHttpClient.java:35)
at com.slack.api.util.http.SlackHttpClient.buildSlackHttpClient(SlackHttpClient.java:31)
at com.slack.api.Slack.<init>(Slack.java:75)
at com.slack.api.Slack.getInstance(Slack.java:89)
at com.fpd.teamcity.slack.SlackGateway.<init>(SlackGateway.scala:137)
at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:142)
... 44 more 
```

--- a/slackIntegration-server/src/main/scala/com/fpd/teamcity/slack/SlackGateway.scala
+++ b/slackIntegration-server/src/main/scala/com/fpd/teamcity/slack/SlackGateway.scala
@@ -109,15 +109,23 @@ object SlackGateway {
       .orElse(getIntProperty("https.proxyPort"))
       .getOrElse(80)
 
-    val proxyUrl = proxyHost.map(host => s"http://$host:$proxyPort")
+    val proxyUrl = proxyHost.map(host => s"$host:$proxyPort")
 
-    proxyUrl.foreach(x => config.setProxyUrl(x))
+    proxyUrl.foreach(x => config.setProxyUrl(s"http://$x"))
+
+    val proxyLogin = getStringProperty("slack.proxyLogin")
+      .orElse(getStringProperty("http.proxyLogin"))
+      .orElse(getStringProperty("https.proxyLogin"))
+
+    val proxyPassword = getStringProperty("slack.proxyPassword")
+      .orElse(getStringProperty("http.proxyPassword"))
+      .orElse(getStringProperty("https.proxyLogin"))
 
     for {
       url <- proxyUrl
-      proxyLogin <- getStringProperty("https.proxyLogin")
-      proxyPassword <- getStringProperty("https.proxyPassword")
-    } yield config.setProxyUrl(s"http://$proxyLogin:$proxyPassword@$url")
+      login <- proxyLogin
+      password <- proxyPassword
+    } yield config.setProxyUrl(s"http://$login:$password@$url")
 
     config
   }

--- a/slackIntegration-server/src/test/scala/com/fpd/teamcity/slack/SlackGatewayTest.scala
+++ b/slackIntegration-server/src/test/scala/com/fpd/teamcity/slack/SlackGatewayTest.scala
@@ -81,6 +81,21 @@ class SlackGatewayTest
     config.getProxyUrl shouldEqual "http://proxy.com:1234"
   }
 
+  "SlackGateway.prepareConfig" should "return correct config if both http and https proxy settings is set" in {
+    System.setProperty("http.proxyHost", "http-proxy.com")
+    System.setProperty("http.proxyPort", "1234")
+    System.setProperty("http.proxyLogin", "http-user")
+    System.setProperty("http.proxyPassword", "http-pass")
+
+    System.setProperty("https.proxyHost", "https-proxy.com")
+    System.setProperty("https.proxyPort", "5678")
+    System.setProperty("https.proxyLogin", "https-user")
+    System.setProperty("https.proxyPassword", "https-pass")
+    val config = SlackGateway.prepareConfig
+
+    config.getProxyUrl shouldEqual "http://http-user:http-pass@http-proxy.com:1234"
+  }
+
   "SlackGateway" should "be instantiated with proxy" in {
     System.setProperty("slack.proxyHost", "proxy.com")
     System.setProperty("slack.proxyPort", "wrong port")

